client/src/Components/Navigation/Menu.js

```diff
@@ -1,101 +1,158 @@
-import React, {Component} from 'react';
+/* eslint-disable jsx-a11y/img-redundant-alt */
+import React, { useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+import { useHistory, Link, useLocation } from "react-router-dom";
+import { Avatar } from "@material-ui/core";
+import decode from "jwt-decode";
 
-export default class Menu extends Component {
-    render() {
-        return (
-            <div>
-                <aside className="main-sidebar sidebar-dark-primary elevation-4">
-                    {/* Brand Logo */}
-                    <a href="index3.html" className="brand-link">
-                        <img src="dist/img/AdminLTELogo.png" alt="AdminLTE Logo" className="brand-image img-circle elevation-3" style={{ opacity: '.8' }} />
-                        <span className="brand-text font-weight-light">TRRADER</span>
-                    </a>
-                    {/* Sidebar */}
-                    <div className="sidebar">
-                        {/* Sidebar user panel (optional) */}
-                        <div className="user-panel mt-3 pb-3 mb-3 d-flex">
-                            <div className="image">
-                                <img src="dist/img/user2-160x160.jpg" className="img-circle elevation-2" alt="User Image" />
-                            </div>
-                            <div className="info">
-                                <a href="#" className="d-block">User</a>
-                            </div>
-                        </div>
-                        {/* SidebarSearch Form */}
-                        <div className="form-inline">
-                            <div className="input-group" data-widget="sidebar-search">
-                                <input className="form-control form-control-sidebar" type="search" placeholder="Search" aria-label="Search" />
-                                <div className="input-group-append">
-                                    <button className="btn btn-sidebar">
-                                        <i className="fas fa-search fa-fw" />
-                                    </button>
-                                </div>
-                            </div>
-                        </div>
-                        {/* Sidebar Menu */}
-                        <nav className="mt-2">
-                            <ul className="nav nav-pills nav-sidebar flex-column" data-widget="treeview" role="menu" data-accordion="false">
-                                {/* Add icons to the links using the .nav-icon class
+const Menu = () => {
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const location = useLocation();
+  const [user, setUser] = useState(JSON.parse(localStorage.getItem("profile")));
+  useEffect(() => {
+    const token = user?.token;
+    if (token) {
+      const decodedToken = decode(token);
+      if (decodedToken.exp * 1000 < new Date().getTime()) logout();
```

Comment:
looks like the `isTokenExpired` check is happening in multiple places. Maybe that should be split out into a separate function :)

Reply:
umm you didn't take my suggestion on the `isTokenExpired` util function